### PR TITLE
Make Debug print all lines to STDERR, and use it for reports of Fatal

### LINKED ISSUE #1560: Fatal stack trace not shown on standard error
I am writing a new binding for `beamer.cls` and am running latexml in order to debug any issues with it. 
However as of the logging rewrite, latexml no longer writes any fatal stack traces to standard error, but only to a log file.

For example:
```
$ latexml --verbose  ./test/examples/beamer.tex
latexml (LaTeXML version 0.8.5; revision f6456924) processing ./test/examples/beamer.tex ...
Fatal:die:/Users/twiesing/perl5/lib/perl5/LaTeXML/Package/beamer.cls.ltxml File /Users/twiesing/perl5/lib/perl5/LaTeXML/Package/beamer.cls.ltxml had an error: at beamer.cls.ltxml
Conversion complete: 1 fatal error (See /Users/twiesing/Projects/github.com/tkw1536/LaTeXML/beamer.latexml.log) (reqd. 2.06s)
```
In order to debug such an error, I would need to know the exact stack trace. 
This can currently only be found in the log file, which makes development a lot slower than before. 
I now need to open the log file every time I want to debug something. 
This also disables all my terminal support for ctrl+clicking on error lines to open them directly in an editor.

There should be some method of showing the stack trace directly on standard error, because neither `--verbose` or `--debug` did anything.

### PR DESCRIPTION
**Pull request by brucemiller**
as it says....

Fixes #1560  (hopefully)

### CODE REVIEW

**Comment by dginev** [comment]
Sadly this won't fix #1560, but is otherwise a nice PR worth merging.

For #1560 you need to _add_ a new `Debug` statement in LaTeXML.pm, but you can't since the print to the `$LOG` happens already in Fatal(). So any half-hearted patch would move the problem from no prints on STDERR, to two prints in `$LOG`. Which is why i mentioned `NoteSTDERR` in the issue, but that feels so patchy I don't want to even add it. 

I'd rather make sure the `Fatal()` subroutine prints to STDERR directly, and the `die` it calls is not used for anything except a status code. That's a separate refactor unrelated to this PR.

LGTM!

**Comment by brucemiller** [comment]
Whether it fixes the exact scenario that @tkw1536 mentioned in #1560 may depend on exactly what that scenario was.  He said it didn't print the "stack trace" to STDERR. And when I encountered a similar situation, I only got the single "FATAL" line, but no details or stack trace on STDERR, only in the log. This patch does indeed fix that situation so that I get the same full message on STDERR and the log.

Since you seem to be talking about a different scenario and hedged saying "hopefully".  In fact, I was surprised to be reminded that `Fatal` didn't print to STDERR, but relied on `die` to do it. Seems like it would make sense to do both full prints in Fatal, and then let the die message be something like "LaTeXML died" (and avoiding recursion). But does that mess something else up?  Can you characterize the scenario where you're getting nothing on STDERR? Would be good to fix correctly while we're thinking about it.

**Comment by dginev** [comment]
Sure just compare `latexml` vs `latexmlc` on any base call with a Fatal say this one:
```
latexmlc 'literal:\makeatletter\def\letters{a}\def\letterloop{\edef\thisname{\csname\letters\endcsname} \the\thisname \g@addto@macro\letters{a} \letterloop}\letterloop'
```

When the Fatal calls `die()` the content of that gets flushed to STDERR in the usual `latexml` case. In my `latexmlc` case it doesn't currently. And rather than writing a custom patch in LaTeXML.pm that tries to fish out exactly where in that `die()` one finds the Fatal message, I see we're in an agreement that it would be much better for the Fatal sub itself to do the print to all available logging streams, pretty much immediately after it assembles the `$message`.

**Comment by brucemiller** [comment]
Ah, hadn't caught the latexml vs latexmlc distinction (though Tom said latexml). Also, I must have gotten desensitized, as Fatal's already show twice in the latexml's log.

**Comment by tkw1536** [comment]
I actually tried both `latexml` and `latexmlc`, but figured reporting the issue for one of them should be enough.

**Comment by dginev** [comment]
Yes, exactly, all executables need to have a wrapping eval for the Fatal `die()` to be consistently handled. If they use LaTeXML.pm, that's handled there, if they don't - they need to add it in separately.

**Comment by brucemiller** [comment]
OK, cool; will pursue that line of thought.

File: lib/LaTeXML.pm [hunk 1]
```diff
@@ -114,7 +114,7 @@ sub initialize_session {
   ## NOTE: This will give double errors, if latexml has already handled it!
   $$latexml{state}->noteStatus('fatal') if $latexml && $@;    # Fatal Error?
   if ($@) {                                                   #Fatal occured!
-    Note($@);
+    Debug($@);
     Note("Initialization complete: " . $latexml->getStatusMessage . ". Aborting.") if defined $latexml;
     # Close and restore STDERR to original condition.
     $$self{log} .= $self->flush_log;
```
[the remark this answers is not in the document]
Comment (brucemiller) [reply lib/LaTeXML.pm:120]: agreed

**Comment by dginev** [comment]
Note to self when I get the time: the new `LaTeXML died!` message gets picked up by LaTeXML.pm during some timeouts, and may need some code to ignore reporting it in that particular code path.

Here is [an example preview from a Fatal doc](https://corpora.mathweb.org/preview/sandbox%5Farxiv%5F1905/tex%5Fto%5Fhtml/1905.01178)

**Pull request merged**